### PR TITLE
implement month navigation and test it

### PR DESCRIPTION
**Pull request by Arkadyus**
This PR lets the user navigate between months in the monthly calendar view using two arrows around the month and year

### CODE REVIEW

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CalendarActivityTest.java [hunk 1]
```diff
@@ -151,4 +152,20 @@ public void monthlyViewOpensRecyclerView() throws InterruptedException {
...
         onView(withText("title")).inRoot(isDialog()).check(matches(isDisplayed()));
     }
+
+    @Test
+    public void monthNavigationWorks() throws InterruptedException {
+        String yearMonthText = YearMonth.now().plusMonths(1).format(DateTimeFormatter.ofPattern("LLLL uuuu"));
+        String yearMonthText2 = YearMonth.now().format(DateTimeFormatter.ofPattern("LLLL uuuu"));
+        // The wait is here in order to wait for the refreshing of the calendar
+        onView(withId(R.id.calendar_screen_view_change)).perform(click());
+        Thread.sleep(1000);
```
Comment (Oxydeas) [inline app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CalendarActivityTest.java:162]: Summoning the god of thread.sleeps @sansive

**Review by Oxydeas — APPROVED** [review]
Good small change in terms of user perspective. Clean code.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/main_activities/CalendarActivity.java [hunk 2]
```diff
@@ -46,13 +49,26 @@ protected void onCreate(Bundle savedInstanceState) {
         findViewById(R.id.calendar_screen_view_change).setOnClickListener(v -> {
             calendarAdapter = calendar.rotateCalendarView(this, calendarAdapter, calendarEvents);
             calendarAdapter.refreshCalendarView(this, currentHouse, "refreshCalendar:failureToRefresh", calendar.getView() == Calendar.CalendarView.MONTHLY);
+            yearMonth.setText(YearMonth.now().format(DateTimeFormatter.ofPattern("LLLL uuuu")));
```
Comment (aindreias) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/panels/main_activities/CalendarActivity.java:52]: How is this pattern represented in the app? (LLLL uuuu)

File: app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/MonthlyAdapter.java [hunk 3]
```diff
@@ -93,6 +102,19 @@ private void prepareMonthlyView(MonthlyDayViewHolder holder, int position) {
         }
     }
 
+    @SuppressLint("NotifyDataSetChanged")
+    public void moveMonth(int monthsDifference, TextView yearMonthText) {
```
Comment (aindreias) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/MonthlyAdapter.java:106]: I think you can use `plusMonths` for both cases, it says [here](https://docs.oracle.com/javase/8/docs/api/java/time/YearMonth.html#plusMonths-long-) that it's allowed? I'm not sure if this is a better option though, so feel free to keep the code as-is.

**Review by aindreias — CHANGES_REQUESTED** [review]
Thanks for the feature! Clean code, just had 1 question and 1 remark - approving once they're addressed 👍

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/main_activities/CalendarActivity.java (hunk 2 above)
Comment (Arkadyus) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/panels/main_activities/CalendarActivity.java:52]: Its the title on top of the monthly calendar view (May 2022 for example), LLLL means the month and uuuu is the year

File: app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/MonthlyAdapter.java (hunk 3 above)
Comment (Arkadyus) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/MonthlyAdapter.java:106]: Oh I just assumed it wasnt possible as both methods exist, ill try to change it in the next commit then, thanks for the remark !

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CalendarActivityTest.java (hunk 1 above)
Comment (sansive) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CalendarActivityTest.java:162]: Thanks! I'll change it on my PR

**Pull request merged**